src/main/webapp/creatives/maps/SimidMap.js

```diff
@@ -0,0 +1,241 @@
+        /** 
+         * The element from the document where the travel method selector lives
+         * @private {?Object}
+         */  
+        this.travelMethodElement_ = null;
+        /** 
+         * The element from the document where the time display div lives
+         * @private {?Object}
+         */   
+        this.timeDisplayElement_ = null;
+    }
+
+    setTravelMethodElement(element) {
+        this.travelMethodElement_ = element;
+    }
+
+    setTimeDisplayElement(element) {
+        this.timeDisplayElement_ = element;
+    }
+
+    setSearchQuery(query) {
+        this.searchQuery_ = query;
+    }
+
+    setMarkerImage(markerImage) {
+        this.markerImage_ = markerImage
+    }
+
+    setCurrentLocation(coordinates = new google.maps.LatLng(DEFAULT_MAP_LAT, DEFAULT_MAP_LNG)) {
+        this.currentLocation_ = coordinates;
+    }
+
+    getMap() {
+        return this.map_;
+    }
+
+    /**
+     * Loads a map object that currently defaults to a hardcoded location.
+     * @param {!Object} element The div within the main document where the map is to be displayed.
+     * @public
+     */
+    displayMap(element) {
+        this.map_ = new google.maps.Map(element, {
+            zoom: DEFAULT_ZOOM,
+            center: this.currentLocation_
+        });
+        this.findNearby_();
+    }
+
+    /**
+     * Searches for the closest corresponding locations based off of the given search parameter,
+     * and places pins on the map that represent the closest locations.
+     * @private
+     */
+    findNearby_() {
+        const request = {
+            location: this.currentLocation_,
+            name: this.searchQuery_,
+            openNow: true,
+            rankBy: google.maps.places.RankBy.DISTANCE
+        };
+        const placeService = new google.maps.places.PlacesService(this.map_);
+        placeService.nearbySearch(request, this.displayResults_.bind(this));
+    }
+
+    /**
+     * Displays the closest advertisement's locations to a user's current location.
+     * @param {!Object} results An array of Place Results from the search query.
+     * @param {!google.maps.places.PlacesServiceStatus} status The status returned 
+     *  by the PlacesService on the completion of its searches.
+     * @private 
+     */
+    displayResults_(results, status) {
+        if (status == google.maps.places.PlacesServiceStatus.OK) {
+            //Active location is set to the closest location to start.
+            this.activeLocation_ = results[0].geometry.location;
+            for (let i = 0; i < DEFAULT_LOCATION_NUM_DISPLAYED; i++) {
+                this.placeMapMarker_(results[i]);
+            }
+            this.displayDirections_();
+        } else {
+            const statusErrorMessage = {
+                message: "ERROR: Failed to complete search: " + status,
+            };
+            //this.simidProtocol.sendMessage(CreativeMessage.LOG, statusErrorMessage);
+            this.playAd_();
```

Comment:
Another idea is to have the owner of this class pass in an error function.  So at any point you could call the error function and the maps will hide and just return to the ad.

Dan's method is fine too.  Keep what you do consistent with what clients of this class should expect from it.

Reply:
Fixed in our meeting